src/CMakeLists.txt

```diff
@@ -95,14 +97,49 @@ set(BOX2D_HEADER_FILES
 	../include/box2d/b2_world_callbacks.h
 	../include/box2d/box2d.h)
 
-add_library(box2d STATIC ${BOX2D_SOURCE_FILES} ${BOX2D_HEADER_FILES})
-target_include_directories(box2d PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}/../include)
-target_include_directories(box2d PRIVATE ${CMAKE_CURRENT_SOURCE_DIR})
+add_library(box2d ${BOX2D_SOURCE_FILES} ${BOX2D_HEADER_FILES})
+
+target_include_directories(box2d
+  PUBLIC
+    ${CMAKE_CURRENT_SOURCE_DIR}/../include
+    $<BUILD_INTERFACE:${CMAKE_CURRENT_BINARY_DIR}>
+  PRIVATE
+    ${CMAKE_CURRENT_SOURCE_DIR}
+)
+
 set_target_properties(box2d PROPERTIES
 	CXX_STANDARD 11
     CXX_STANDARD_REQUIRED YES
     CXX_EXTENSIONS NO
+    VERSION ${PROJECT_VERSION}
+    SOVERSION ${PROJECT_VERSION_MAJOR}
+)
+
+generate_export_header(box2d
+  BASE_NAME b2
+  EXPORT_MACRO_NAME B2_API
+  EXPORT_FILE_NAME b2_api.h
+  STATIC_DEFINE B2_STATIC
 )
 
+if (NOT BUILD_SHARED_LIBS)
+  target_compile_definitions(box2d
+    PUBLIC
+      B2_STATIC
+  )
+endif()
+
+target_compile_definitions(box2d
+  PRIVATE
+    box2d_EXPORTS
```

Comment:
CMake defines this by default when you compile a shared lib.

Reply:
Right!